cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/ProgramScheduleStoreDataset.java

```diff
@@ -499,6 +519,24 @@ public ProgramScheduleRecord getScheduleRecord(ScheduleId scheduleId) throws Not
     return result;
   }
 
+  private List<ProgramScheduleRecord> listScheduleRecords(ApplicationId appId, @Nullable ProgramId programId) {
```

Comment:
I would add a Javadoc for this method, just so you can explain why `programId` can be null.